### PR TITLE
Fix for chunked requests

### PR DESCRIPTION
**Pull request by coorasse**

### CODE REVIEW

File: actionpack/lib/action_dispatch/http/request.rb [hunk 1]
```diff
@@ -278,12 +278,21 @@ def server_software
...
+          else
+            raw_post_body.read(content_length)
+          end
         raw_post_body.rewind if raw_post_body.respond_to?(:rewind)
       end
       @env['RAW_POST_DATA']
     end
 
+    def chunked?
+      headers['Transfer-Encoding'].try(:downcase) == 'chunked'
```
Comment (kaspth) [inline actionpack/lib/action_dispatch/http/request.rb]: Could also do `headers['Transfer-Encoding'] =~ /chunked/i`

Comment (coorasse) [inline actionpack/lib/action_dispatch/http/request.rb]: maybe `!!(headers['Trasfer-Encoding'] =~ /chunked/i)` ?

**Comment by coorasse** [comment]
Inspired from: https://github.com/xaviershay/rails/commit/46e7202f8129f4f18d164c1be314c15b6ce975a6.

It fixes #15079. 

What about security? Is it safe to just call `read` on line 283?

File: actionpack/lib/action_dispatch/http/request.rb (hunk 1 above)
Comment (kaspth) [inline actionpack/lib/action_dispatch/http/request.rb]: At the Rails organization we like relying on Ruby's semantics and don't particularly care for constructs like `!!`. So if we can make due without doing that then :+1:

**Comment by yoongkang** [comment]
Probably needs a test.

**Comment by maclover7** [comment]
Please rebase.

**Pull request closed without merging**